### DISCUSSION TITLE
Оптимизация расчетов в таблице

### DISCUSSION THREAD

**Discussion by Spyker086** [totumonline/totum-ru-issues-and-discussions#577] [Как это сделать?] [answered]
Всем привет! 

Подскажите. Чет сам не нашел решения. Есть простая таблица.
В одном из столбцов в коде рассчитываю значение поля. И в результат беру listCount.
В условном форматировании у меня работает тот же код, который выдает в comment, listJoin того же списка.

Проблема в том что сам расчет тяжелый и рассчитывается получается два раза.
Как мне промежуточный результат расчета  (сам список получается) в поле Код, передать в поле "Условное форматирование". Дабы не считать два раза.

Есть решение, чтоб сам список отдельно считать в отдельном поле и в нужном столбце брать уже результат. Но задался вопросом: можно ли сделать без доп столбца?

**Comment by ttm-alexeypolunin** [comment]
Hi. Рассчитать его в коде идущем первым по ходу и передать результат в процессную переменную (поищите поиском ее в доках).

**Comment by Spyker086** [comment, marked as answer]
Когда в коде формирую проц переменную, а потом ее использую в условном форматировании, то у меня во всем столбце значение "comment" одинаковое получается (в зависимости какую строку пересчитал последнюю). Нужно как то значение из кода передать в условное форматирование в рамках одной строки, а не всего столбца.

**Reply by ttm-alexeypolunin** [reply]
Надо формировать переменную составляя ее name с id строки. Сколько строк, столько и переменных создается

**Reply by Spyker086** [reply]
Последний вопрос ))
Переменную формирую так: procVar(name: #id; value: "222"; default: $#lc)

А вот как к ней обратиться $@'#id' ? Чтоб id строки не явно указывать?

**Reply by ttm-alexeypolunin** [reply]
В этом случае надо использовать вызов через функцию procVar(name: #id; default: $#lc)

когда нет параметра value это вызов, когда есть запись
дефаулт нужен именно при вызове, при записи он не используется

**Reply by Spyker086** [reply]
Супер! Все получилось! Спасибо!
